include/wasp/convert/to_text.h

```diff
@@ -0,0 +1,133 @@
+//
+// Copyright 2020 WebAssembly Community Group participants
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+#ifndef WASP_CONVERT_TO_TEXT_H_
+#define WASP_CONVERT_TO_TEXT_H_
+
+#include <map>
+#include <memory>
+#include <string>
+#include <vector>
+
+#include "wasp/base/at.h"
+#include "wasp/base/buffer.h"
+#include "wasp/base/optional.h"
+#include "wasp/binary/types.h"
+#include "wasp/text/types.h"
+
+namespace wasp::convert {
+
+// TODO: Rename
+struct TextContext {
+  text::Text Add(string_view);
+
+  // TODO: The buffers need to have stable pointers (for use by string_view or
+  // span), so for now they are all std::unique_ptr<T>. This could be optimized
+  // in a number of ways, however (e.g. allocating in larger blocks of chars,
+  // and appending to that).
+  std::vector<std::unique_ptr<std::string>> strings;
+};
+
+// Helpers.
+auto ToText(TextContext&, const At<binary::HeapType>&) -> At<text::HeapType>;
```

Comment:
It still seems weird to write auto here .. because its not auto at all right?  The type is specifically declared at the end of the line?

Reply:
Yeah, this is a C++11 feature. I found that it made it a bit easier to read since the function names line up.